specs/CoreWebView2.CreateFromCOMObject.md

```diff
@@ -47,8 +53,52 @@ public class MyWebView2Control
 }
 ```
 
+## WinRT to COM
+
+```c++
+winrt::com_ptr<ICoreWebView2> GetComICoreWebView2FromCoreWebView2(
+    winrt::Microsoft::Web::WebView2::Core::CoreWebView2 coreWebView2WinRT)
+{
+    // Get the COM interop interface from the WinRT CoreWebView2.
+    auto interop = coreWebView2WinRT.as<ICoreWebView2Interop2>();
+
+    // Get the COM ICoreWebView2 object from the COM interop interface.
+    winrt::com_ptr<ICoreWebView2> coreWebView2Com;
+    winrt::check_hresult(interop->GetComICoreWebView2(coreWebView2Com.put()));
```

Comment:
Why is one direction "Create" and the other "Get"?

Answering my own question: .NET CoreWebView2 and WinRT CoreWebView2 are both wrappers around the underlying COM ICoreWebView2, so going from CoreWebView2 to ICoreWebView2 is a "Get" (there is always an inner ICoreWebView2). But going from ICoreWebView2 to CoreWebView2 requires that a wrapper be manufactured, hence the Create.

(I was surprised that they are wrappers. I though the COM ICoreWebView2 directly implemented the WinRT methods too - you just had to QI for them.)

Reply:
> Answering my own question: ...

You are correct. Any concerns?

> (I was surprised that they are wrappers. I though the COM ICoreWebView2 directly implemented the WinRT methods too - you just had to QI for them.)

WinRT is a wrapper because we previously supported Win7 where various WinRT things wouldn't be available. A year or two ago we've dropped support for Win7+8 but have no justification to go back and change the implementation of our WinRT support.